includes/OAuth/OAuthAuthenticator.php

```diff
@@ -62,15 +60,20 @@ public function setAuthenticationEndpoint(string $endpoint)
      */
     public function authenticate(string $clientId, string $clientSecret): object
     {
-        $response = $this->client->post($this->authenticationEndpoint, [
-            "form_params" => [
-                "grant_type" => "client_credentials",
-                "client_id" => $clientId,
-                "client_secret" => $clientSecret
-            ]
-        ]);
-
+        $timestampCreated = time();
+        try {
+            $response = $this->client->post($this->authenticationEndpoint, [
+                "form_params" => [
+                    "grant_type" => "client_credentials",
+                    "client_id" => $clientId,
+                    "client_secret" => $clientSecret
+                ]
+            ]);
+        } catch (\Exception $e) {
+            throw new \Exception('Authentication failed: could not connect to the authentication host.');
+        }
         if ($response->getStatusCode() === 400) {
+            echo "400er";
```

Comment:
Probabbly a leftover from debugging.

```suggestion
            echo "400er";
```